tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,639 @@
+    )
+    inputs = TrainingInput(s3_data=input_path)
+
+    instance_count = ParameterInteger(name="InstanceCount", default_value=1)
+    instance_type = ParameterString(name="InstanceType", default_value="ml.m5.xlarge")
+    good_enough_input = ParameterInteger(name="GoodEnoughInput", default_value=1)
+
+    pytorch_estimator = PyTorch(
+        entry_point=entry_point,
+        role=role,
+        framework_version="1.5.0",
+        py_version="py3",
+        instance_count=instance_count,
+        instance_type=instance_type,
+        sagemaker_session=pipeline_session,
+    )
+    step_train = TrainingStep(
+        name="pytorch-train",
+        estimator=pytorch_estimator,
+        inputs=inputs,
+    )
+    model = Model(
+        image_uri=pytorch_estimator.training_image_uri(),
+        model_data=step_train.properties.ModelArtifacts.S3ModelArtifacts,
+        sagemaker_session=pipeline_session,
+        role=role,
+    )
+    regis_model_step_args = model.register(
+        content_types=["*"],
+        response_types=["application/json"],
+        inference_instances=["ml.m5.xlarge"],
+        transform_instances=["ml.m5.xlarge"],
+        description="test-description",
+    )
+    step_model_regis = ModelStep(
+        name="pytorch-register-model",
+        model=model,
+        step_args=regis_model_step_args,
+    )
+    create_model_step_args = model.create(
+        instance_type="ml.m5.large",
+        accelerator_type="ml.eia1.medium",
+    )
+
+    step_model_create = ModelStep(
+        name="pytorch-model",
+        model=model,
+        step_args=create_model_step_args,
+    )
+
+    step_cond = ConditionStep(
+        name="cond-good-enough",
+        conditions=[
+            ConditionGreaterThanOrEqualTo(left=good_enough_input, right=1),
+        ],
+        if_steps=[step_model_regis],
+        else_steps=[step_model_create],
+        depends_on=[step_train],
+    )
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[
+            good_enough_input,
+            instance_count,
+            instance_type,
+        ],
+        steps=[step_train, step_cond],
+        sagemaker_session=pipeline_session,
+    )
+    try:
+        pipeline.create(role)
+        execution1 = pipeline.start(parameters={})
+        execution2 = pipeline.start(parameters={"GoodEnoughInput": 0})
+        try:
+            execution1.wait(delay=30, max_attempts=60)
+        except WaiterError:
+            pass
+        execution1_steps = execution1.list_steps()
+        assert len(execution1_steps) == 3
+        for step in execution1_steps:
+            assert step["StepStatus"] == "Succeeded"
+
+        try:
+            execution2.wait(delay=30, max_attempts=60)
+        except WaiterError:
+            pass
+        execution2_steps = execution2.list_steps()
+        assert len(execution2_steps) == 3
+        for step in execution2_steps:
+            assert step["StepStatus"] == "Succeeded"
```

Comment:
can we also assert that ModelStepMetadata is non-empty here?

Reply:
Updated in the new commit